miden-crypto/src/merkle/smt/full/large/mod.rs

```diff
@@ -0,0 +1,1112 @@
+use alloc::{boxed::Box, vec::Vec};
+use core::mem;
+use std::sync::Arc;
+
+use num::Integer;
+use rayon::prelude::*;
+
+use super::{
+    EMPTY_WORD, EmptySubtreeRoots, InnerNode, InnerNodeInfo, InnerNodes, LeafIndex, Leaves,
+    MerkleError, MerklePath, MutationSet, NodeIndex, Rpo256, RpoDigest, SMT_DEPTH, Smt, SmtLeaf,
+    SmtProof, SparseMerkleTree, Word,
+    concurrent::{
+        MutatedSubtreeLeaves, PairComputations, SUBTREE_DEPTH, SubtreeLeaf, SubtreeLeavesIter,
+        build_subtree, fetch_sibling_pair, process_sorted_pairs_to_leaves,
+    },
+};
+use crate::merkle::smt::{NodeMutation, NodeMutations, UnorderedMap};
+
+#[cfg(test)]
+mod tests;
+
+mod subtree;
+use subtree::Subtree;
+
+mod storage;
+#[cfg(feature = "rocksdb")]
+pub use storage::RocksDbStorage;
+pub use storage::{MemoryStorage, SmtStorage};
+
+// CONSTANTS
+// ================================================================================================
+
+const IN_MEMORY_DEPTH: u8 = 24;
+
+// TYPES
+// ================================================================================================
+
+// LargeSmt
+// ================================================================================================
+
+/// Sparse Merkle tree mapping 256-bit keys to 256-bit values. Both keys and values are represented
+/// by 4 field elements.
+///
+/// All leaves sit at depth 64. The most significant element of the key is used to identify the leaf
+/// to which the key maps.
+///
+/// A leaf is either empty, or holds one or more key-value pairs. An empty leaf hashes to the empty
+/// word. Otherwise, a leaf hashes to the hash of its key-value pairs, ordered by key first, value
+/// second.
```

Comment:
This documentation should mention what distinguishes it from the other Sparse Merkle tree types. To my understanding all the points mentioned here also apply to `Smt`.